<launch>
  <!-- Dataset dirs -->
  <arg name="frame_num" default="99999" />
  <arg name="start_num" default="0" />


  <arg name="dataset_dir" default="$(optenv DATASET_DIR /)" />
  <arg name="dataset_dir_real" default="/media/jhz/fish/SLAM/myData/ICL_NUIM/traj2n_frei_png/" />

  <arg name="dataset_class" default="" />
  <arg name="dataset_id" default="" />

  <arg name="dataset_dir_real_class_id" default="$(arg dataset_dir_real)/$(arg dataset_class)/$(arg dataset_id)/" />

  <arg name="ellipsoid_slam_dir" default="$(optenv EllipsoidSLAM_PATH)" />

  <!-- Publish rgb/depth images -->
  <node name="tool_publish_images" pkg="EllipsoidSLAM" type="tool_publish_images" args="$(arg dataset_dir_real_class_id) $(arg frame_num) $(arg start_num)" >
    <remap from="/camera/rgb" to="/camera/rgb/image_raw" />
  </node>
  <node name="pub_odom_withTimeStamp" pkg="EllipsoidSLAM" type="pub_odom_withTimeStamp" args="$(arg dataset_dir_real_class_id)/FrameTrajectory.txt" />
  <node name="pub_bbox_withTimeStamp" pkg="EllipsoidSLAM" type="pub_bbox_withTimeStamp" args="$(arg dataset_dir_real_class_id)/bbox/" />

  <!-- Core Node -->
  <!--
  <node name="ros_ellipsoid_slam_real" pkg="EllipsoidSLAM" type="ros_ellipsoid_slam_real" args="$(arg ellipsoid_slam_dir)/Example/param/kinect2.yaml" output="screen" >
        <remap from="/camera/rgb/image" to="/camera/rgb/image_raw"/>
        <remap from="/camera/depth/image" to="/camera/depth/image_raw"/>

  </node>
  -->

  
</launch>